pkg/status/combinedstatus-resolution.go

```diff
@@ -827,3 +886,15 @@ func valueEqual(a, b *v1alpha1.Value) bool {
 		return false
 	}
 }
+
+func parseNumber(s string) (float64, error) {
+	if intValue, err := strconv.Atoi(s); err == nil {
+		return float64(intValue), nil
```

Comment:
`float64(intValue)` can lose precision. How about returning an `any` that holds either an `int64`, a `uint64` that is not `==` any `int64`, or a `float64`?